projects/client/RabbitMQ.Client/src/client/impl/ModelBase.cs

```diff
@@ -1233,6 +1246,50 @@ public void BasicCancel(string consumerTag)
                 body);
         }
 
+        public void BasicBatchPublish(string exchange,
+    string routingKey,
+    bool mandatory,
+    IEnumerable<BatchMessage> messages)
+        {
+            foreach (var message in messages)
+            {
+                if (message.basicProperties == null)
+                {
+                    message.basicProperties = CreateBasicProperties();
+                }
+
+                if (NextPublishSeqNo > 0)
+                {
+                    lock (m_unconfirmedSet.SyncRoot)
```

Comment:
Would it change anything if we would acquire all the sequence numbers in one go inside a lock instead of reacquiring the lock for each message?